### PR TITLE
GCC*: import from homebrew/versions

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Homebrew/versions PR: https://github.com/Homebrew/homebrew-versions/pull/1472

If any of these don't build they will likely be 💥. I've skipped the older GCC versions that are either unused or barely used from analytics. I'm trying to depend on the latest versions of the various GCC libraries to see if we can avoid importing them but on the inevitable failures will pull in stuff like `gmp4` where needed.

### CODE REVIEW

File: Formula/gcc@4.6.rb [hunk 1]
```diff
@@ -0,0 +1,209 @@
+class Gcc46 < Formula
```
Comment (dunn) [inline Formula/gcc@4.6.rb]: `GccAT46` and so forth.

Comment (MikeMcQuaid) [inline Formula/gcc@4.6.rb]: Cheers, done.

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
:tada:

**Pull request closed without merging**